nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -70,12 +83,30 @@ class MotionModel
     // }
 
     const bool is_holo = isHolonomic();
+    float max_delta_vx = model_dt_ * control_constraints_.ax_max;
+    float min_delta_vx = model_dt_ * control_constraints_.ax_min;
+    float max_delta_vy = model_dt_ * control_constraints_.ay_max;
+    float max_delta_wz = model_dt_ * control_constraints_.az_max;
+    float vx_last = state.vx(0, 0);
+    float vy_last = state.vy(0, 0);
+    float wz_last = state.wz(0, 0);
     for (unsigned int i = 0; i != state.vx.shape(0); i++) {
       for (unsigned int j = 1; j != state.vx.shape(1); j++) {
-        state.vx(i, j) = state.cvx(i, j - 1);
-        state.wz(i, j) = state.cwz(i, j - 1);
+        float & cvx_curr = state.cvx(i, j - 1);
+        cvx_curr = std::clamp(cvx_curr, vx_last + min_delta_vx, vx_last + max_delta_vx);
+        state.vx(i, j) = cvx_curr;
+        vx_last = cvx_curr;
```

Comment:
The vx need to be rest between the nested for loops, not just before the loop. Otherwise we start with `vx_last` for sample `N+1` as the last vx for sample `N` rather than being the first sample of `N+1`